operator_rolling_upgrades.md

```diff
@@ -0,0 +1,54 @@
+# Operator Rolling Upgrades
+Optionally allow a Infinispan CR cluster to be upgraded with zero downtime and no loss of data.
+
+## Spec
+Add a field to the CR spec that determines whether an upgrade should be `rolling` or `gracefulShutdown`. The latter is the current approach that does not allow data migration.
```

Comment:
I'm the only one confused by the usage of `rolling`? It conflict with the k8s rolling upgrade procedure in my mind

Reply:
I think that's fair, it's a very overloaded term.

How about `hotrodMigration`? That fits with the new name for this capability "Hot Rod Migration"

So eventually we would have:

```yaml
spec:
  upgrade: hotrodMigration | inline | gracefulShutdown
```

Reply:
How about `migration` only, `hotrod` sounds a technical aspect imo. But it's just a minor detail, thery're both ok to me.